api/v1alpha1/storageclaim_types.go

```diff
@@ -53,7 +53,7 @@ type StorageClientNamespacedName struct {
 
 // StorageClaimSpec defines the desired state of StorageClaim
 type StorageClaimSpec struct {
-	//+kubebuilder:validation:Enum=block;sharedfile
+	//+kubebuilder:validation:XValidation:rule="self.lowerAscii()=='block'||self.lowerAscii()=='sharedfile'",message="value should be either 'sharedfile' or 'block' (case insensitive)"
```

Comment:
Let's not confuse the user with unnecessary options in our messages, it will just land a hand for abuse.  The fact that it is case insensitive does not mean we need to publicly announce it as such

```suggestion
	//+kubebuilder:validation:XValidation:rule="self.lowerAscii()=='block'||self.lowerAscii()=='sharedfile'",message="value should be either 'sharedfile' or 'block'"
```

Reply:
makes sense, done.